app/src/test/java/com/codesoom/controllers/ProductControllerWebTest.java

```diff
@@ -0,0 +1,266 @@
+package com.codesoom.controllers;
+
+import com.codesoom.application.ProductService;
+import com.codesoom.domain.Product;
+import com.codesoom.exception.ProductNotFoundException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(ProductController.class)
+@AutoConfigureMockMvc
+@DisplayName("ProductController Web 테스트")
+class ProductControllerWebTest {
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ProductController productController;
+
+    @MockBean
+    private ProductService productService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private static final String PRODUCT_NAME = "고양이 장남감";
+    private static final String PRODUCT_MAKER = "코드숨";
+    private static final BigDecimal PRODUCT_PRICE = BigDecimal.valueOf(10000);
+    private static final String PRODUCT_IMAGE_URL = "test.jpg";
+
+
+    @Nested
+    @DisplayName("GET 요청은")
+    class Describe_get {
+        @Nested
+        @DisplayName("등록된 Product가 있다면")
+        class Context_has_product {
+            final int productCount = 3;
+
+            @BeforeEach
+            void setUp() {
+                List<Product> products = new ArrayList<>();
+
+                for (int i = 1; i < productCount; i++) {
+                    productController.create(getProduct(i));
+                    products.add(getProduct(i));
+                }
+                given(productService.getProducts()).willReturn(products);
+            }
+
+            @Test
+            @DisplayName("전체 리스트를 리턴한다.")
+            void it_return_list() throws Exception {
+                mockMvc.perform(get("/products")
+                                .accept(MediaType.APPLICATION_JSON_UTF8))
```

Comment:
`APPLICATION_JSON_UTF8`은 deprecated된 값입니다. `APPLICATION_JSON_UTF8`의 JavaDoc을 읽어보세요. JavaDoc을 읽어보면 얘를 쓰지 말고 어떤 것을 써야 하는지도 알 수 있을 거에요.

Reply:
>f 5.2 in favor of APPLICATION_JSON since major browsers like Chrome now >comply with the specification  and interpret correctly UTF-8 special >characters without requiring a charset=UTF-8 parameter.

주요 브라우저에서는 UTF-8 문자를 알아서 해석하네요! 

테스트에서 인코딩이 깨지는 현상이 있어서 

```java
this.mockMvc = MockMvcBuilders.webAppContextSetup(wac)
                .addFilters(new CharacterEncodingFilter("UTF-8", true))
                .alwaysDo(print())
                .build();
```
인코딩해주는 코드를 테스트 실행 전에 실행하도록 했습니다!